### PR TITLE
[pt2] add metas for `_pdist_forward` and `_pdist_backward`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #103817
* #103815

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -1362,6 +1362,31 @@ def meta_pad3d_backward(grad_output, input, padding):
...
+@out_wrapper()
+def meta__pdist_forward(self: Tensor, p: float = 2) -> Tensor:
+    check(self.is_contiguous(), lambda: "_pdist_forward requires contiguous input")
+    n = self.size(0)
+    if n <= 1:
+        return self.new_empty([0]).to(memory_format=torch.legacy_contiguous_format)  # type: ignore[call-overload]
+    else:
+        return self.new_empty((n * (n - 1) // 2,)).to(
+            memory_format=torch.legacy_contiguous_format
+        )  # type: ignore[call-overload]
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:1375]: See `_pdist_forward` in `pytorch/aten/src/ATen/native/Distance.cpp`.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**